### PR TITLE
Add OpenGraph Support

### LINKED ISSUE #708: Add rich preview metadata
This is an example

https://github.com/sphinx-doc/sphinx/issues/2645

### PR DESCRIPTION
**Pull request by Daltz333**
Closes #708

### CODE REVIEW

File: source/conf.py [hunk 1]
```diff
@@ -36,10 +36,14 @@
     'sphinxcontrib.ghcontributors',
     'sphinxcontrib.remoteliteralinclude',
     'sphinxcontrib.rsvgconverter',
+    'sphinxext.opengraph',
     'hoverxref.extension',
     'notfound.extension'
 ]
 
+# Configure OpenGraph support
+og_site_url = 'https://docs.wpilib.org/en/latest/'
```
Comment (TheTripleV) [inline source/conf.py:45]:
```suggestion
ogp_site_url = 'https://docs.wpilib.org/en/latest/'
```

**Comment by sciencewhiz** [comment]
Right now, with this PR, it isn't enough for chiefdelphi to make a preview using the OpenGraph data (but I'm not sure what is necessary).

File: source/conf.py [hunk 2]
```diff
@@ -36,10 +36,16 @@
...
     'sphinxcontrib.rsvgconverter',
+    'sphinxext.opengraph',
     'hoverxref.extension',
     'notfound.extension'
 ]
 
+# Configure OpenGraph support
+ogp_site_url = 'https://frc-docs--711.org.readthedocs.build/en/711/'
+ogp_site_name = 'FIRST Robotics Competition Documentation'
+ogp_image = "https://raw.githubusercontent.com/wpilibsuite/frc-docs/master/source/assets/wpilibDocsLogo.png"
```
Comment (AustinShalit) [inline source/conf.py:47]:
```suggestion
ogp_image = "https://docs.wpilib.org/en/latest/_static/wpilibDocsLogo.png"
```

**Comment by sciencewhiz** [comment]
The image has a transparent background. It would probably be best to have a non-transparent version for this, so the white text in WPI and FRC doesn't get lost.

**Pull request merged**